### PR TITLE
overhaul netrw's decompress

### PR DESCRIPTION
**Pull request by Konfekt**
addresses [0]

[0]: https://github.com/vim/vim/issues/16099

### CODE REVIEW

**Comment by saccarosium** [comment]
`..` is the syntax for concatenating strings in vim9script with legacy vimscript it should be use `.` instead.

File: runtime/autoload/netrw.vim [hunk 1]
```diff
@@ -365,7 +365,35 @@ call s:NetrwInit("g:netrw_cygdrive","/cygdrive")
...
+            \ .."'.zst':      'zstd -T0 -d',"
+            \ .."'.Z':        'uncompress -k',"
+            \ .."'.tar':      'tar -xvf',"
+            \ .."'.tar.bz':   'tar -xvjf',"
+            \ .."'.tar.bz2':  'tar -xvjf',"
+            \ .."'.tbz':      'tar -xvjf',"
+            \ .."'.tbz2':     'tar -xvjf',"
+            \ .."'.tar.gz':   'tar -xvzf',"
+            \ .."'.tgz':      'tar -xvzf',"
+            \ .."'.tar.lzma': '"..(has('unix')?"XZ_OPT=-T0":(has("win32") && &shell =~? '\vcmd(\.exe)?$' ?"setx XZ_OPT=-T0 &&":"")).." tar -xvf --lzma',"
```
Comment (saccarosium) [inline runtime/autoload/netrw.vim]: Can you do the check for unix or windows once and than concatenate it? Also a bit of spacing in the concatenate would be nice... It is really hard to read.

**Review by saccarosium — CHANGES_REQUESTED** [review]

**Comment by Konfekt** [comment]
> .. is the syntax for concatenating strings in vim9script with legacy vimscript it should be use . instead.

I asked @chrisbra for it once, but it's been around for long enough

**Comment by 00-kat** [comment]
The docs for `g:netrw_decompress` should probably be updated to show the new default too.

**Comment by chrisbra** [comment]
thanks!

**Pull request closed without merging**